substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jdk/Resources.java

```diff
@@ -276,30 +397,43 @@ public static Enumeration<URL> createURLs(String moduleName, String resourceName
             return null;
         }
 
+        boolean missingMetadata = true;
+
         List<URL> resourcesURLs = new ArrayList<>();
         String canonicalResourceName = toCanonicalForm(resourceName);
         boolean shouldAppendTrailingSlash = hasTrailingSlash(resourceName);
         /* If moduleName was unspecified we have to consider all modules in the image */
         if (moduleName == null) {
             for (Module module : BootModuleLayerSupport.instance().getBootLayer().modules()) {
-                ResourceStorageEntry entry = Resources.get(module.getName(), resourceName);
+                Object entry = Resources.get(module.getName(), resourceName, false);
+                if (entry == MISSING_METADATA) {
+                    continue;
+                }
+                missingMetadata = false;
                 addURLEntries(resourcesURLs, entry, module.getName(), shouldAppendTrailingSlash ? canonicalResourceName + '/' : canonicalResourceName);
             }
         }
-        ResourceStorageEntry explicitEntry = Resources.get(moduleName, resourceName);
-        addURLEntries(resourcesURLs, explicitEntry, moduleName, shouldAppendTrailingSlash ? canonicalResourceName + '/' : canonicalResourceName);
+        Object explicitEntry = Resources.get(moduleName, resourceName, false);
+        if (explicitEntry != MISSING_METADATA) {
+            missingMetadata = false;
+            addURLEntries(resourcesURLs, explicitEntry, moduleName, shouldAppendTrailingSlash ? canonicalResourceName + '/' : canonicalResourceName);
+        }
+
+        if (missingMetadata) {
+            throw MissingResourceMetadataException.missingResource(resourceName);
+        }
 
         if (resourcesURLs.isEmpty()) {
             return Collections.emptyEnumeration();
         }
         return Collections.enumeration(resourcesURLs);
     }
 
-    private static void addURLEntries(List<URL> resourcesURLs, ResourceStorageEntry entry, String moduleName, String canonicalResourceName) {
+    private static void addURLEntries(List<URL> resourcesURLs, Object entry, String moduleName, String canonicalResourceName) {
         if (entry == null) {
             return;
         }
-        int numberOfResources = entry.getData().size();
+        int numberOfResources = ((ResourceStorageEntry) entry).getData().size();
```

Comment:
If this is unconditional, you can probably change the parameter type of this function